tests/phpunit/integration/Services/ManifestStoreTest.php

```diff
@@ -41,6 +66,221 @@ private function createFetcherMock( string $yaml, bool $success = true ): MockOb
+		$status1 = $store->get();
+		$this->assertTrue( $status1->isOK(), 'First get() should succeed' );
+		$data1 = $status1->getValue();
+		$this->assertFalse( $data1['from_cache'], 'First call should not be from cache' );
+		$hash1 = $data1['meta']['hash'];
+		
+		// Second call - should return cached data (same lastFetched = 1000)
+		$status2 = $store->get();
+		$this->assertTrue( $status2->isOK(), 'Second get() should succeed' );
+		$data2 = $status2->getValue();
+		$this->assertTrue( $data2['from_cache'], 'Second call should be from cache' );
+		$this->assertEquals( $hash1, $data2['meta']['hash'] );
+		
+		// Update last_fetched timestamp (simulating a repository refresh)
+		$this->repoRegistry->updateRepoEntry( $repoId, [
+			'last_fetched' => 2000,
+		] );
+		
+		// Verify timestamp was updated
+		$updatedRepo = $this->repoRegistry->getRepo( $repoId );
+		$this->assertEquals( 2000, $updatedRepo->lastFetched() );
+		
+		// Third call - lastFetched changed to 2000, should fetch fresh data
+		// (different cache key due to different lastFetched)
+		$status3 = $store->get();
+		$this->assertTrue( $status3->isOK(), 'Third get() should succeed' );
+		$data3 = $status3->getValue();
+		$this->assertFalse( 
+			$data3['from_cache'], 
+			'Cache should be invalidated when last_fetched timestamp changes'
+		);
+		$this->assertEquals( $hash1, $data3['meta']['hash'] );
+	}
+
+	/**
+	 * Test that different last_fetched timestamps use different cache keys.
+	 *
+	 * Two ManifestStore instances for the same repo/ref but with different
+	 * last_fetched timestamps should not share cache entries.
+	 */
+	public function testCacheKeyIncludesLastFetchedTimestamp(): void {
+		$yaml = <<<YAML
+schema_version: '1.0.0'
+packs:
+  test-pack:
+    version: '1.0.0'
+YAML;
+
+		$repoUrl = 'https://github.com/example/test-repo-2';
+		$ref = 'main';
+		
+		// Create first repository with last_fetched = 1000
+		$repoId1 = $this->repoRegistry->ensureRepoEntry( $repoUrl, [
+			'last_fetched' => 1000,
+		] );
+		
+		/** @var ManifestFetcher&MockObject $fetcher */
+		$fetcher = $this->createFetcherMock( $yaml );
+		
+		$store1 = new ManifestStore( $repoUrl, $ref, null, $fetcher, $this->repoRegistry );
+		
+		// Cache data for store1 (lastFetched = 1000)
+		$status1 = $store1->get();
+		$this->assertFalse( $status1->getValue()['from_cache'] );
+		
+		// Update last_fetched to 2000
+		$this->repoRegistry->updateRepoEntry( $repoId1, [
+			'last_fetched' => 2000,
+		] );
+		
+		// Create a new store instance (will see updated last_fetched = 2000)
+		$store2 = new ManifestStore( $repoUrl, $ref, null, $fetcher, $this->repoRegistry );
+		
+		// store2 should NOT have cached data because it uses a different cache key
+		// (lastFetched = 2000 vs 1000)
+		$status2 = $store2->get();
+		$this->assertFalse( 
+			$status2->getValue()['from_cache'],
+			'Different last_fetched timestamps should use different cache keys'
+		);
+	}
```

Comment:
test doesn't test anything unique - `get()` recomputes the hash key every time, so store1 should also have the new cache key if the row in the repository has been updated, however that is not tested here: after the update, store1 is not used.